packages/salesforce-adapter/src/filters/territory.ts

```diff
@@ -32,13 +43,46 @@ const removeCustomFieldsFromTypes = (elements: Element[], typeNames: string[]):
     })
 }
 
+const isTerritoryRelatedChange = (change: Change): change is Change<InstanceElement> => (
+  isInstanceChange(change) && isAdditionOrModificationChange(change)
+   && territory2Types.some(typeName => isInstanceOfTypeChange(typeName)(change))
+)
+
+const setTerritoryDeployPkgStructure = (elements: InstanceElement[]): void => {
+  elements.forEach(element => {
+    const nestedDirsNames = [parentApiName(element)]
+    const { suffix } = element.type.annotations
+    let name = apiName(element)
+    if (territory2TypesToNestedDirName[metadataType(element)] !== undefined) {
+      const nestedDirName = territory2TypesToNestedDirName[metadataType(element)]
+      nestedDirsNames.push(nestedDirName)
+      name = apiName(element, true)
+    }
+    const contentFileName = `${name}${suffix === undefined ? '' : `.${suffix}`}`
+    element.annotate({ [CONTENT_FILENAME_OVERRIDE]: [...nestedDirsNames, contentFileName] })
+  })
+}
+
 const filterCreator: FilterCreator = () => ({
   onFetch: async elements => {
     // Territory2 and Territory2Model support custom fields - these are returned
     // in a CustomObject with the appropriate name and also in each instance of these types
     // We remove the fields from the instances to avoid duplication
     removeCustomFieldsFromTypes(elements, [TERRITORY2_TYPE, TERRITORY2_MODEL_TYPE])
   },
+
+  // territory2 types require a special deploy pkg structure (SALTO-1200)
+  preDeploy: async changes => {
+    const territoryInstances = changes.filter(isTerritoryRelatedChange).map(getChangeElement)
+    setTerritoryDeployPkgStructure(territoryInstances)
+  },
+  onDeploy: async changes => {
+    const territoryInstances = changes.filter(isTerritoryRelatedChange).map(getChangeElement)
+    territoryInstances.forEach(elem => {
+      elem.annotations = _.omit(elem.annotations, [CONTENT_FILENAME_OVERRIDE])
+    })
```

Comment:
style nit
```suggestion
    changes
      .filter(isTerritoryRelatedChange)
      .map(getChangeElement)
      .forEach(elem => {
        delete elem.annotations[CONTENT_FILENAME_OVERRIDE]
      })
```

similar thing can be done in `preDeploy` if we change `setTerritoryDeployPkgStructure` to get a single element instead of a list, it could be
```typescript
changes
  .filter(isTerritoryRelatedChange)
  .map(getChangeElement)
  .forEach(setTerritoryDeployPkgStructure)
```